kstreams/prometheus/monitor.py

```diff
@@ -132,11 +132,24 @@ def clean_stream_consumer_metrics(self, stream: Stream) -> None:
             for topic_partition in topic_partitions:
                 topic = topic_partition.topic
                 partition = topic_partition.partition
-                self.MET_LAG.remove(topic, partition, group_id)
-                self.MET_POSITION_LAG.remove(topic, partition, group_id)
-                self.MET_COMMITTED.remove(topic, partition, group_id)
-                self.MET_POSITION.remove(topic, partition, group_id)
-                self.MET_HIGHWATER.remove(topic, partition, group_id)
+
+                metrics_found = False
+                for sample in self.MET_LAG.collect()[0].samples:
```

Comment:
my impression is that it won't fail, since the MET_LAG is a gauge that is created by the monitor instantiation. and the [0] just grabs the element out of it's list holder. even with no metrics, the met_lag will exist will zero labels but the metric is still in [0]. that's my impression. indeed. worst case, we introduce a new patch. no one is using this feature but me. I will merge.